File: easy/q1/q1.java
package easy.q1;

import java.util.HashMap;
import java.util.Map;

class q1 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numIndices = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (numIndices.containsKey(complement)) {
                return new int[]{numIndices.get(complement), i};
            }

            numIndices.put(nums[i], i);
        }

        return new int[]{-1, -1};
    }
}

File: easy/q1/q1.cpp
#include <unordered_map>
#include <vector>
using namespace std;
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        unordered_map<int, int> numIndices;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];

            if (numIndices.find(complement) != numIndices.end()) {
                return {numIndices[complement], i};
            }

            numIndices[nums[i]] = i;
        }

        return {-1, -1};
    }
};
